src/infomgmag/mars/CountryAgent.java

```diff
@@ -84,9 +97,59 @@ public boolean bordersEnemy(){  //checks if a territory borders an enemy
         return bordersenemies;
     }
 
-    public double ownWholeContinent() {
+    public boolean ownWholeContinent(){     //checks if the agents owns the whole continent except this territory
```

Comment:
This is overly complex; instead you could just loop through the territories, and return false if you don't have it, and return true at the end. The best way to do it would be to use `territory.getBelongsTo().stream().allMatch(x -> x.getOwner() == this.mars)` or something like that.

Reply:
Done